mono/metadata/metadata.c

```diff
@@ -2438,7 +2420,8 @@ delete_image_set (MonoImageSet *set)
 	for (i = 0; i < set->nimages; ++i)
 		set->images [i]->image_sets = g_slist_remove (set->images [i]->image_sets, set);
 
-	g_ptr_array_remove (image_sets, set);
+	if (image_sets)
```

Comment:
This can't happen, the list must be created for an image set to exist.

Do you have an example of this not being true?